runtime/compiler/env/PersistentAllocator.cpp

```diff
@@ -228,58 +324,256 @@ void
+   size_t const index = freeBlocksIndex(block->size());
    TR_ASSERT(index != LARGE_BLOCK_LIST_INDEX, "freeFixedSizeBlock should be used for small blocks, so index cannot be LARGE_BLOCK_LIST_INDEX");
-   block->_next = _freeBlocks[index];
+   block->setNext(_freeBlocks[index]);
    _freeBlocks[index] = block;
    }
 
 void
 PersistentAllocator::freeVariableSizeBlock(Block * block)
    {
    // Appropriate lock should have been obtained
-   TR_ASSERT(block->_size > 0, "Block size is non-positive");
-   block->_next = NULL;
+   TR_ASSERT(block->size() > 0, "Block size is non-positive");
+   block->setNext(NULL);
    // Add block to the variable-size-block chain which is in ascending size order.
    //
-   TR_ASSERT(freeBlocksIndex(block->_size) == LARGE_BLOCK_LIST_INDEX, "freeVariableSizeBlock should be used for large blocks, so index should be LARGE_BLOCK_LIST_INDEX");
+   TR_ASSERT(freeBlocksIndex(block->size()) == LARGE_BLOCK_LIST_INDEX, "freeVariableSizeBlock should be used for large blocks, so index should be LARGE_BLOCK_LIST_INDEX");
    Block * blockIterator = _freeBlocks[LARGE_BLOCK_LIST_INDEX];
-   if (!blockIterator || !(blockIterator->_size < block->_size) )
+   if (!blockIterator || !(blockIterator->size() < block->size()) )
       {
-      block->_next = _freeBlocks[LARGE_BLOCK_LIST_INDEX];
+      block->setNext(_freeBlocks[LARGE_BLOCK_LIST_INDEX]);
       _freeBlocks[LARGE_BLOCK_LIST_INDEX] = block;
       }
    else
       {
-      while (blockIterator->next() && blockIterator->next()->_size < block->_size)
+      while (blockIterator->next() && blockIterator->next()->size() < block->size())
          {
          blockIterator = blockIterator->next();
          }
-      block->_next = blockIterator->next();
-      blockIterator->_next = block;
+      block->setNext(blockIterator->next());
+      blockIterator->setNext(block);
+      }
+   }
+
+#if defined(J9VM_OPT_JITSERVER)
+size_t 
+PersistentAllocator::getInterval(size_t blockSize)
+   {  
+   // Find the power-of-two interval that this block size belongs to
+   TR_ASSERT(blockSize >= PERSISTANT_BLOCK_SIZE_BUCKETS * sizeof(void *), "getInterval should be used only on big blocks. blockSize=%zu", blockSize);
+   // If very large block
+   if (blockSize >= (1 << (BITS_TO_SHIFT_FIRST_INTERVAL + NUM_INTERVALS - 1)))
+      return NUM_INTERVALS - 1; // last one
+   blockSize >>= (BITS_TO_SHIFT_FIRST_INTERVAL + 1);
+   // log2 implementation that uses a lookup table of 16 entries
+   // The value to compute the log on cannot be larger than 1 byte
+   static_assert(NUM_INTERVALS <= 9, "For large values of NUM_INTERVALS the log2 implementation below will not work");
+   TR_ASSERT(blockSize < (1 << 8), "blockSize is too large %zu to apply the log2 implementation below", blockSize);
+   static const uint8_t _logTable[16] = { 0, 0, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3 };
+   size_t upper;
+   return (upper = blockSize >> 4) ? 4 + _logTable[upper] : _logTable[blockSize];
```

Comment:
`floorLog2` from the header file is not very efficient because it uses recursion and can recurse 64 times. That function is marked `constexpr` because it needs to be evaluated during the compilation and used as a hardcoded value by the compiler. In c++11 `constexpr` functions need to have only one line, which basically forces one to use recursion.